src/main/java/mate/academy/springboot/datajpa/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package mate.academy.springboot.datajpa.service.impl;
+
+import java.math.BigDecimal;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.repository.ProductRepository;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.stereotype.Service;
+
+@AllArgsConstructor
+@Service
+public class ProductServiceImpl implements ProductService {
+    private final ProductRepository productRepository;
+
+    @Override
+    public Product save(Product product) {
```

Comment:
split into 2 methods: 
* `create()` - which is just like the one you have here, 
* `update()` - where you check the id and use Optional
same for category